tests/testthat/test-estimate_dispersion.R

```diff
@@ -55,79 +58,133 @@ test_that("Basic functionality with valid inputs", {
   )
-  expect_error(estimate_dispersion(valid_nowcasts, non_integer_trunc_rts))
+  expect_error(estimate_dispersion(
+    valid_nowcasts, non_integer_trunc_rts, valid_rts))
 })
 
-### Test 4: Edge Cases ---------------------------------------------------------
-test_that("Edge cases are handled properly", {
+test_that("estimate_dispersion: Edge cases are handled properly", {
   # Empty lists
-  expect_error(expect_warning(estimate_dispersion(list(), list(), n = 0)))
+  expect_error(estimate_dispersion(list(), list(), list(), n = 0))
 
-  # No NAs---------------------------------------------------------------------
+  # No NAs in truncated reporting triangles
   expect_warning(estimate_dispersion(
     valid_nowcasts,
-    lapply(valid_nowcasts, round)
+    lapply(valid_nowcasts, round),
+    valid_rts
   ))
 
-  # NA-filled matrices (This should error!)
+  # NA-filled matrices
   na_nowcasts <- list(matrix(NA, 2, 3), matrix(NA, 1, 3))
   na_trunc <- list(matrix(NA, 2, 3), matrix(NA, 1, 3))
-  expect_error(estimate_dispersion(na_nowcasts, na_trunc, n = 2))
+  expect_error(estimate_dispersion(na_nowcasts, na_trunc, valid_rts, n = 2))
 })
 
-### Test 5: Dimension Validation -----------------------------------------------
-test_that("Matrix dimension validation works", {
+test_that("estimate_dispersion: Matrix dimension validation works", {
   # Mismatched dimensions between nowcasts and trunc_rts
   bad_trunc_rts <- list(
     test_triangle[1:5, ],
     test_triangle[1:3, ]
   )
   expect_error(
-    estimate_dispersion(valid_nowcasts, bad_trunc_rts)
+    estimate_dispersion(valid_nowcasts, bad_trunc_rts, valid_rts)
   )
 })
 
-## Test 6: fit_nb returns NA if nothing passed to it---------------------------
-test_that("Passing in empty vector returns NA", {
+test_that(".fit_nb: Passing in empty vector returns NA", {
   x <- NULL
   NA_result <- .fit_nb(x, mu = 1)
   expect_true(is.na(NA_result))
 })
+
+test_that("estimate_dispersion: Works with ragged reporting triangles", {
+  # Create a triangle with known delay PMF
+  sim_delay_pmf <- c(0.1, 0.2, 0.3, 0.1, 0.1, 0.1)
+
+  # Generate counts for each reference date
+  counts <- c(30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150,
+              160, 170, 180, 190, 200)
+
+  # Create a complete triangle based on the known delay PMF
+  complete_triangle <- lapply(counts, function(x) round(x * sim_delay_pmf))
+  complete_triangle <- do.call(rbind, complete_triangle)
+
+  # Create a reporting triangle with every other day reporting
+  ragged_triangle <- generate_triangle(
+    complete_triangle,
+    structure = 2
+  )
+
+  # Create truncated triangles and retrospective triangles
+  trunc_rts <- truncate_triangles(ragged_triangle)
+  retro_rts <- generate_triangles(trunc_rts, structure = 2)
+
+  # Generate nowcasts from the ragged triangles
+  retro_nowcasts <- generate_pt_nowcast_mat_list(retro_rts)
+
+  # Estimate dispersion parameters
+  disp_params <- estimate_dispersion(
```

Comment:
This all looks good. However, I noticed when testing and comparing the dispersion estimates that you get from this versus the diagonal triangle, that we don't get back all estimates of 999.9 (which we should bc the only error is in the rounding error).

This led me to find a bug in `apply_delay()` that I should have thought about and caught when reviewing #99. Will open a separate issue.